libs/langgraph/langgraph/pregel/loop.py

```diff
@@ -1187,6 +1253,42 @@ def _update_mv(self, key: str, values: Sequence[Any]) -> None:
 
         return self.submit(cast(WritableManagedValue, managed_value).aupdate, values)
 
+    async def amatch_cached_writes(self) -> Sequence[PregelExecutableTask]:
+        if self.cache is None:
+            return []
+        matched: list[PregelExecutableTask] = []
+        if cached := {
+            (t.cache_key.ns, t.cache_key.key): t
+            for t in self.tasks.values()
+            if t.cache_key and not t.writes
+        }:
+            for key, values in (await self.cache.aget(tuple(cached))).items():
+                task = cached[key]
+                task.writes.extend(values)
+                matched.append(task)
+        return matched
+
+    def put_writes(self, task_id: str, writes: WritesT) -> None:
+        """Put writes for a task, to be read by the next tick."""
+        super().put_writes(task_id, writes)
+        if not writes or self.cache is None or not hasattr(self, "tasks"):
+            return
+        task = self.tasks.get(task_id)
+        if task is None or task.cache_key is None:
+            return
+        if writes[0][0] in (INTERRUPT, ERROR):
```

Comment:
this could go  in hte `or` statement above right? 
```python
if not writes or self.cache is None or not hasattr(self, "tasks") or writes[0][0] in (INTERRUPT, ERROR)
```